var/spack/repos/builtin/packages/lapackpp/package.py

```diff
@@ -52,16 +56,34 @@ class Lapackpp(CMakePackage):
     for (lpp_ver, bpp_ver) in _versions:
         depends_on("blaspp@" + bpp_ver, when="@" + lpp_ver)
 
+    depends_on("blaspp ~cuda", when="~cuda")
+    depends_on("blaspp +cuda", when="+cuda")
+    depends_on("blaspp ~rocm", when="~rocm")
```

Comment:
Does `lapackpp~cuda` actually require `blaspp~cuda`,
and `lapackpp~rocm` require `blaspp~rocm`?
Normally that's what I would expect, that both are `+cuda`, or both `+rocm`, or both `~rocm ~cuda`, but I don't think it's required. It could be `lapackpp~rocm~cuda` and (`blaspp+rocm` or `blaspp+cuda`). But those are edge cases I wouldn't expect and wouldn't test, so may be simpler to disallow.

Reply:
I'd say that since we don't test those edge cases and they are unusual, that those cases should be considered unsupported.  It is more likely that someone would want cuda or rocm support turned on for all major components, so this makes it much easier for the user by chaining the gpu config down to the dependencies.